deeplabcut/generate_training_dataset/labeling_toolbox.py

```diff
@@ -867,14 +873,51 @@ def saveEachImage(self):
             self.dataFrame.loc[self.relativeimagenames[self.iter]][
                 self.scorer, bp[0][-2], "y"
             ] = bp[-1][1]
+	
+    def ResetEachImage(self):
+	    """
+	    Reset data for each image
+	    """
+	    for idx, bp in enumerate(self.updatedCoords):
+	        self.dataFrame.loc[self.relativeimagenames[self.iter]][
+	            self.scorer, bp[0][-2], "x"
+	        ] = None
+	        self.dataFrame.loc[self.relativeimagenames[self.iter]][
+	            self.scorer, bp[0][-2], "y"
+	        ] = None
+    
+
+    def deleteImage(self, event):
+    	image_path = os.path.join( self.currentDirectory, self.dataFrame.index[self.iter])
+    	print("Delete Image Path : ", image_path)
+    	os.remove(image_path)
+    	MainFrame.ResetEachImage(self)
+    	#  Checks for the last image and disables the Next button
+    	if len(self.index) - self.iter == 1:
+    		return
+    	self.nextImage(event=None)
+    	return
 
     def saveDataSet(self, event):
         """
         Saves the final dataframe
         """
+
+        # Remove previous save
+        csv_path = os.path.join(self.dir, "CollectedData_" + self.scorer + ".csv") 
+        if os.path.exists( csv_path):
+        	os.remove( csv_path)
+
+        hdf_path = os.path.join(self.dir, "CollectedData_" + self.scorer + ".h5")
+        if os.path.exists( hdf_path):
+        	os.remove( hdf_path)
```

Comment:
Is that necessary, given that data are overwritten by default? I'm worried that if something fails—although I admit this shouldn't happen—between that line and to_csv()/to_hdf(), the user is left with no data.

Reply:
Have replaced it with backup file creation for extra safety, eventually can add restore function as well